### PR TITLE
Feature people further enhancement

### PR DESCRIPTION
**Pull request by Joltrix**
Needed for StarcounterSamples/People#34

Added an int to keep track of which additional columns to display

### CODE REVIEW

File: Ring3/CustomElement/CustomContactInfoRelationType.cs [hunk 1]
```diff
@@ -1,4 +1,6 @@
 ﻿namespace Simplified.Ring3 {
     public class CustomContactInfoRelationType : Simplified.Ring2.ContactInfoRelationType {
+
+        public int SelectedIndex;
```
Comment (warpech) [inline Ring3/CustomElement/CustomContactInfoRelationType.cs]: I think that this is an over-simplification. 

`Ring2` is totally too broad to store information about column layout in People app.

I think that there should be a `PeopleSettings` class that stores this information. similar to this: https://github.com/StarcounterPrefabs/Simplified/blob/master/Ring6/Images/ImagesSettings.cs

**Comment by warpech** [comment]
Code review done, I think that data model needs a change.

**Comment by warpech** [comment]
While you're at it, please rename the directory `Simplified/Ring3/CustomElement` to `Simplified/Ring3/CustomContactInfo`.

**Comment by warpech** [comment]
It is possible to take benefit from the fact that in Starcounter you can keep reference to an actual database object in Settings. 

Something like the following in `Ring6/People/*`:

```cs
class PeopleSetting : Something {
}

class CustomContactInfoColumnSetting : Relation {
  public int Order;
  
  [SynonymousTo(WhatIs)]
  public CustomContactInfoRelationType CustomContactInfoRelationType;

  [SynonymousTo(ToWhat)]
  public PeopleSetting PeopleSetting;
}

/* ---- */

var columnConfigForOrganisationsList = new PeopleSettings() { Name = "COLUMN_CONFIG_FOR_ORGANISATIONS_LIST" };
var col1 = new CustomContactInfoColumnSetting() {
  Order = 0,
  CustomContactInfoRelationType = Db.SQL("SELECT * FROM CustomContactInfoRelationType WHERE Name = ?", "Twitter")
  PeopleSetting = columnConfigForOrganisationsList
}

var columnConfigForPersonsList = new PeopleSettings() { Name = "COLUMN_CONFIG_FOR_PERSONS_LIST" };
var col1 = new CustomContactInfoColumnSetting() {
  Order = 0,
  CustomContactInfoRelationType = Db.SQL("SELECT * FROM CustomContactInfoRelationType WHERE Name = ?", "GitHub")
  PeopleSetting = columnConfigForPersonsList
}
var col2 = new CustomContactInfoColumnSetting() {
  Order = 1,
  CustomContactInfoRelationType = Db.SQL("SELECT * FROM CustomContactInfoRelationType WHERE Name = ?", "Twitter")
  PeopleSetting = columnConfigForPersonsList
}
```

Then, it should be possible to create an interface to pick up columns to display on organisations and persons lists with the following dual list UI:

image

Please note that the logic for such UI should be on the server. We have a list of all unselected custom types on the left, two buttons with handlers in `.json.cs`, a list of all selected custom types on the right.

This is just my two cents. Use it if you like it.

**Comment by Joltrix** [comment]
@warpech 
I have now changed what you've suggested, and I added a Dual List for the selection of which custom fields to display.

**Comment by warpech** [comment]
Are you sure you've pushed everything in People? I get the following error:

2016-09-08 15_43_28-people - microsoft visual studio

**Comment by warpech** [comment]
Looks good! Merging

**Pull request merged**